module/VuFind/tests/unit-tests/src/VuFindTest/UrlShortener/DatabaseTest.php

```diff
@@ -84,14 +84,21 @@ protected function getShortener(
     /**
      * Mock entity plugin manager.
      *
+     * @param int|null $setExpectation Flag to set the method expectations.
+     *
      * @return MockObject
      */
-    protected function getPluginManager()
+    protected function getPluginManager($setExpectation = null)
     {
         $pluginManager = $this->getMockBuilder(
             \VuFind\Db\Entity\PluginManager::class
         )->disableOriginalConstructor()
             ->getMock();
+        if ($setExpectation) {
+            $pluginManager->expects($this->once())->method('get')
```

Comment:
If $setExpectation is an int, maybe it's better to expect `$this->exactly($setExpectation)` instead of `$this->once()` in case you need to expect multiple calls in the future. If you're just using it as a flag, it would be better to use `bool` instead of `int`.